Controllers/ChatController.cs

```diff
@@ -0,0 +1,135 @@
+                ) return View("ConsultantChat", new ConsultantChatViewModel
+                {
+                    ConversationId = consultantChatConnection.ConversationId,
+                    RedirectAction = consultantChatConnection.RedirectAction,
+                    Host = conversation.Host,
+                    Path = conversation.Path
+                });
+
+            if (conversation.Status != ConversationStatus.NEW) return RedirectToAction("NewConversationList", new ModalViewModel
+            {
+                ModalLabel = "Already taken",
+                ModalText = new List<string>() { "Other consultant has already joined to this conversation." },
+                IsVisible = true,
+                ModalType = ModalStyles.ERROR
+            });
+
+            var isSaved = await _chatRepository.AssignConsultantToConversation(conversation,
+                                                                             consultantId,
+                                                                             consultantChatConnection.RowVersion);
+
+            if (!isSaved) return RedirectToAction("NewConversationList", new ModalViewModel
+            {
+                ModalLabel = "Already taken",
+                ModalText = new List<string>() { "Other consultant has already joined to this conversation." },
+                IsVisible = true,
+                ModalType = ModalStyles.ERROR
+            });
+
+
+            return View("ConsultantChat", new ConsultantChatViewModel
+            {
+                ConversationId = consultantChatConnection.ConversationId,
+                RedirectAction = consultantChatConnection.RedirectAction,
+                Host = conversation.Host,
+                Path = conversation.Path
+            });
+        }
+
+        [Authorize(Roles = UserRoleValue.CONSULTANT)]
+        [HttpGet("new-conversation-list")]
+        public async Task<IActionResult> NewConversationList(int pageIndex = 1, ModalViewModel modalViewModel = null)
+        {
+            var newConversations = _chatRepository.GetNewConversationsQuery();
+
+            var newConversationsPaginated = await PaginatedList<Conversation>.CreateAsync(
+                                                                                newConversations,
+                                                                                pageIndex,
+                                                                                10);
+
+            return View(
+                "NewConversationList",
+                    new NewConversationListViewModel
+                    {
+                        ConversationList = newConversationsPaginated,
+                        Modal = modalViewModel
+                    }
+                );
+        }
+
+        [Authorize(Roles = UserRoleValue.CONSULTANT)]
+        [HttpGet("in-progress-conversation-list")]
+        public async Task<IActionResult> InProgressConversationList(int pageIndex = 1)
```

Comment:
I think it's good to check if pageIndex is more than 0, as it could be overrided by user manually I'm guessing.